### PR TITLE
Crawler demo

### PR DESCRIPTION
**Pull request by vsalavatov**

### CODE REVIEW

File: cmd/crawler/main.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    c := colly.NewCollector()
+
+    used := make(map[string]bool)
+    urlQueue := make([]string, 0)
+    articles := make([]string, 0)
+
+    c.OnHTML("a[href]", func(e *colly.HTMLElement) {
+        url := e.Attr("href")
+        if !used[url] && strings.Contains(url, "arxiv.org") {
+            if strings.Contains(url, "abs") {
```
Comment (M0nsteR) [inline cmd/crawler/main.go:41]: You'd better move all your constant strings to a config with some meaningful descriptions (hostname, topic etc).

File: cmd/crawler/main.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    c.OnRequest(func(r *colly.Request) {
+		fmt.Println("Visiting", r.URL)
+	})
+
+    urlQueue = append(urlQueue, "https://arxiv.org/covid19search")
+
+    needArticles := 10
+
+    for len(articles) < needArticles && len(urlQueue) > 0 {
```
Comment (M0nsteR) [inline cmd/crawler/main.go]: Looks weird to me.. try to express this intent in a more idiomatic and clear way.

File: cmd/crawler/main.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    if err != nil {
+        panic(err)
+    }
+    defer f.Close()
+    w := bufio.NewWriter(f)
+
+    fmt.Fprintf(w, "Id;Title;Authors;Abstract;LastUpdateTimestamp\n")
+
+    for i, url := range articles {
+        fmt.Println("Checking article number", i)
```
Comment (M0nsteR) [inline cmd/crawler/main.go]: A separate func might be in order here.

**Review by M0nsteR — CHANGES_REQUESTED** [review]
Readability needs to be improved.

File: cmd/crawler/main.go (hunk 1 above)
Comment (peltorator) [reply cmd/crawler/main.go:41]: Done

File: cmd/crawler/main.go (hunk 2 above)
Comment (peltorator) [reply cmd/crawler/main.go]: Done

File: cmd/crawler/main.go (hunk 3 above)
Comment (peltorator) [reply cmd/crawler/main.go]: Done

**Review by M0nsteR — APPROVED** [review]
Thanx. Good to go.

**Pull request merged**